SECP-service/src/main/resources/assets/app/js/chat/chat-controller.js

```diff
@@ -2,7 +2,7 @@
 
 angular.module('SECP')
   .controller('ChatController',
-    function ($scope, $modal, Chat, Socket) {
+    function ($scope, $modal, Chat, Socket, Admin, $q) {
```

Comment:
I'm not sure if you need a better name for $q. Also, I thought you needed to be an Admin to use Admin.js service. @tosinoni can you confirm this?

Reply:
The $q is a built in angular variable used for promises.

The Admin.js service was created to talk to the admin resource on the server side. So, technically only admins should use it. The getAllFilters should probably be in the chatService but it's fine here.

Reply:
👍